### PR TITLE
Move both examples into single Dockerfile for Docker Hub (fixes #347)

### LINKED ISSUE #347: Push the simple and random examples to docker hub
It'd be nice to be able to install those two examples with docker invocations so both an instrumented daemon and prometheus can be started this way. This would allow a full end to end example with two `docker run` and with a config file bound mounted, which seems fairly minimal.

The labels could be:
- `prometheus/example-golang-simple`
- `prometheus/example-golang-random`

I can prepare a PR building the docker images, but would be good to know if people are happy with the idea.

### PR DESCRIPTION
**Pull request by pedrosland**
This PR should be the last step to get the examples on the Docker Hub. It:

* merges the Dockerfile for the two examples into one
* puts it in the root of the repo for Docker Hub
* updates to Go 1.10.3 to build the examples

The two commands to run the examples with this change could be:
```sh
docker run -ti prom/golang-examples /simple
docker run -ti prom/golang-examples /random
```

I have tested this on the hub and it does build:
```sh
docker run -ti -p 8080:8080 pedrosland/prom-test:test /simple
```

See issue #347 and the discussion on #348.

Please let me know @beorn7 if you would like any changes. I can also make a PR to the docs repo to add this to the getting started guide.

### CODE REVIEW

**Comment by beorn7** [comment]
This appears to me exactly the thing we need. However, since my Docker foo is still mediocre, it would be great if somebody else could chime in for a sanity check, for example @discordianfish . :pray:

File: Dockerfile [hunk 1]
```diff
@@ -4,17 +4,20 @@
 #    docker build -f examples/$name/Dockerfile -t prometheus/golang-example-$name .
 
 # Builder image, where we build the example.
-FROM golang:1.9.0 AS builder
+FROM golang:1.10.3 AS builder
```
Comment (grobie) [inline Dockerfile]: Golang 1 has a stable and guaranteed interface, using `golang:1` would remove the need to constantly update this version with every new release.

Comment (beorn7) [reply Dockerfile]: If that works, great.

Comment (pedrosland) [reply Dockerfile]: Thanks for the feedback. Sounds good. I've updated the PR.

**Comment by discordianfish** [comment]
LGTM

**Comment by beorn7** [comment]
Thanks everybody.

**Pull request merged**